packages/walt-compiler/src/tokenizer/comments/index.js

```diff
@@ -2,17 +2,53 @@
 import token from "../token";
 import Syntax from "../../Syntax";
 
-const everything = () => everything;
+import {
+  SLASH,
+  SINGLE_LINE,
+  MULTI_LINE_START,
+  MULTI_LINE_END,
+  COMMENT_IDENTIFIERS,
+  everything,
+} from "./constants";
 
-const slash = char => {
-  if (char === "/") {
-    return everything;
+let isMultiline: boolean = false;
+let previous: string;
+
+const isComment = (current: string) => {
+  let value = null;
+  switch (`${previous}${current}`) {
+    case MULTI_LINE_END: {
+      isMultiline = false;
+      value = isComment;
+      break;
+    }
+    case MULTI_LINE_START: {
+      isMultiline = true;
+      break;
+    }
+    case SINGLE_LINE: {
+      value = everything;
+      break;
+    }
+  }
+
+  if (isMultiline) {
+    value = isComment;
+    previous = current;
```

Comment:
The if block here, it can just go into the `switch/case` above.

Reply:
Agreed